cmd/test/main_test.go

```diff
@@ -240,6 +252,49 @@ func test_testCommand_version_flags_validation(t *testing.T, ctx *TestCommandCon
 	assert.NoError(t, err)
 }
 
+func test_testCommand_empty_dir(t *testing.T, readerMock *ReaderMock, ctx *TestCommandContext) {
+	emptyDir := t.TempDir()
+	emptyDirPaths := filepath.Join(emptyDir, "*.yaml")
+
+	readerMock.On("FilterFiles", []string{emptyDirPaths}).Return([]string{}, nil)
+	err := executeTestCommand(ctx, []string{emptyDirPaths})
+
+	assert.EqualError(t, err, "No files detected")
+	readerMock.AssertCalled(t, "FilterFiles", []string{emptyDirPaths})
+}
+
+func test_testCommand_no_valid_k8s_resources(t *testing.T, readerMock *ReaderMock, evaluator *mockEvaluator, k8sValidator *K8sValidatorMock, filesConfigurations []*extractor.FileConfigurations, printerMock *PrinterMock, ctx *TestCommandContext) {
+	testPath := pathFromRoot("internal/fixtures/nonKube/*.(yaml|json)")
+	readerMock.On("FilterFiles", []string{testPath}).Return([]string{testPath}, nil)
+	err := executeTestCommand(ctx, []string{testPath})
+
+	assert.NotNil(t, err)
+	assert.EqualError(t, err, "")
+	readerMock.AssertCalled(t, "FilterFiles", []string{testPath})
+	k8sValidator.AssertCalled(t, "ValidateResources", mock.Anything, 100)
+	evaluator.AssertCalled(t, "CreateEvaluation", mock.Anything, mock.Anything, mock.Anything, mock.Anything)
+	evaluator.AssertCalled(t, "Evaluate", filesConfigurations, mock.Anything, mock.Anything)
+	printerMock.AssertCalled(t, "PrintWarnings", mock.Anything)
+	printerMock.AssertCalled(t, "PrintEvaluationSummary", mock.Anything, mock.Anything)
+	printerMock.AssertCalled(t, "PrintSummaryTable", mock.Anything)
+}
+
+func test_testCommand_multiple_resources(t *testing.T, readerMock *ReaderMock, evaluator *mockEvaluator, k8sValidator *K8sValidatorMock, filesConfigurations []*extractor.FileConfigurations, printerMock *PrinterMock, ctx *TestCommandContext) {
+	testPath := pathFromRoot("internal/fixtures/kube/pass-all.yaml")
+	readerMock.On("FilterFiles", []string{testPath}).Return([]string{testPath}, nil)
+
+	err := executeTestCommand(ctx, []string{testPath})
+
+	assert.NoError(t, err)
+	readerMock.AssertCalled(t, "FilterFiles", []string{testPath})
+	k8sValidator.AssertCalled(t, "ValidateResources", mock.Anything, 100)
+	evaluator.AssertCalled(t, "CreateEvaluation", mock.Anything, mock.Anything, mock.Anything, mock.Anything)
+	evaluator.AssertCalled(t, "Evaluate", filesConfigurations, mock.Anything, mock.Anything)
+	printerMock.AssertCalled(t, "PrintWarnings", mock.Anything)
+	printerMock.AssertCalled(t, "PrintEvaluationSummary", mock.Anything, mock.Anything)
+	printerMock.AssertCalled(t, "PrintSummaryTable", mock.Anything)
```

Comment:
Here I think we should test that all files pass K8s schema and yaml schema as well as our policy check validation.

Reply:
I had added `assert.NoError(t, err)` to test if the command with parameter `internal/fixtures/kube/pass-all.yaml` is overall passing or not. I didn't find a way to explicitly test the passing of K8s schema, passing of yaml schema and passing of policy checks.

So, my approach is to execute the command with the path and see if it returns any error or not. That way we can confirm that all files are passing.

But there is a pitfall in this approach ( I think you are also saying about this in the above comment), I am not testing how many files are tested against `datree` checks. It would be better if the test could check the report given by the datree command (i.e. the test summary, no. of files tested etc.) and see if all files are tested by matching the no. of files evaluated with the actual no. of files.

But again I didn't find any way to get that report. Do you have any suggestion how to address that issue?

Reply:
I understand. To test with what exactly a function was called with don't use `mock.Anything`, this basically means "be called with anything just I just want to ensure you have been called". Instead, I suggest creating a struct with your test case. I recommend you to review `cliClient_test` and use it as a reference. Feel free to reach out in our slack channel for more information :)

Reply:
thanks @noaabarki , this was my doubt. Thanks for clarifying :)

Reply:
:))